File: eval/fixtures/php-taskflow/src/Store/Task.php
<?php

declare(strict_types=1);

namespace App\Store;

use DateTimeImmutable;

/** A task entity: identity, title, priority, optional due date, done flag. */
class Task
{
    public bool $done = false;

    public function __construct(
        public int $id,
        public string $title,
        public Priority $priority = Priority::Normal,
        public ?DateTimeImmutable $due = null,
    ) {
    }

    /** A task is overdue when it has a due date in the past and is not done. */
    public function overdue(DateTimeImmutable $now): bool
    {
        if ($this->done || $this->due === null) {
            return false;
        }
        return $this->due < $now;
    }

    public function markDone(): void
    {
        $this->done = true;
    }
}

File: eval/fixtures/python-taskflow/taskflow/store/task.py
"""A task entity: identity, title, priority, optional due date, done flag."""

from datetime import datetime
from typing import Optional

from taskflow.store.priority import Priority


class Task:
    def __init__(
        self,
        id: int,
        title: str,
        priority: Priority = Priority.NORMAL,
        due: Optional[datetime] = None,
    ) -> None:
        self.id = id
        self.title = title
        self.priority = priority
        self.due = due
        self.done = False

    def overdue(self, now: datetime) -> bool:
        """A task is overdue when it has a due date in the past and is not done."""
        if self.done or self.due is None:
            return False
        return self.due < now

    def mark_done(self) -> None:
        self.done = True
